Source/SwiftyDropbox/Platform/SwiftyDropbox_iOS/OAuthMobile.swift

```diff
@@ -358,12 +401,10 @@ open class MobileSharedApplication: SharedApplication {
     }
 
     open func dismissAuthController() {
-        if let controller = self.controller {
-            if let presentedViewController = controller.presentedViewController {
-                if presentedViewController.isBeingDismissed == false && presentedViewController is MobileSafariViewController {
-                    controller.dismiss(animated: true, completion: nil)
-                }
-            }
+        // Only MobileSafariViewController needs to be manually dismissed
+        if let presentedViewController = self.authChannel as? MobileSafariViewController, !presentedViewController.isBeingDismissed {
+            self.authChannel = nil
+            presentedViewController.dismiss(animated: true, completion: nil)
         }
     }
```

Comment:
dismissAuthController() is called after you log in, the sessions properly dismiss themselves